apps/chat-e2e/src/testData/conversationHistory/conversationData.ts

```diff
@@ -215,17 +215,18 @@ export class ConversationData extends FolderData {
       conversation,
       replayIndex,
     );
+    const conversationCopy = JSON.parse(
+      JSON.stringify(conversation),
+    ) as Conversation;
     const partialAssistantMessage = replayIndex
-      ? conversation.messages[replayIndex + 2]
-      : conversation.messages.findLast((m) => m.role === 'assistant');
+      ? conversationCopy.messages[replayIndex + 2]
```

Comment:
Consider adding validation for replayIndex to ensure it's within the bounds of the messages array

Reply:
added validation into `fillReplayData()` method